client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts

```diff
@@ -39,9 +39,12 @@ export class WebSocketTransport implements ITransport {
 
             webSocket.onclose = (event: CloseEvent) => {
                 // webSocket will be null if the transport did not start successfully
-                if (this.webSocket && (event.wasClean === false || event.code !== 1000)) {
-                    if (this.onError) {
-                        this.onError(event);
+                if (this.onClosed && this.webSocket) {
+                    if (event.wasClean === false || event.code !== 1000) {
+                        this.onClosed(new Error(`Reason: ${event.reason}. Code: ${event.code}`));
```

Comment:
Can we put a clearer textual description in there too? Something like `WebSocket connection terminated with status code: ${event.code} - ${event.reason}`